CONTRIBUTING.md

```diff
@@ -22,12 +22,23 @@ And run the test suite (should be all green with pending tests):
 
 If you have problems installing nmatrix, please consult the [nmatrix installation wiki](https://github.com/SciRuby/nmatrix/wiki/Installation) or the [mailing list](https://groups.google.com/forum/#!forum/sciruby-dev).
 
+**NOTE**: `Daru` is compatible with Ruby versions < 2.5; for later Ruby versions it breaks, returning the following error when `bundle exec rspec` is run for Ruby 
+versions >= 2.5.
+
+```
+bundler: failed to load command: rspec (/home/.rbenv/versions/2.5.1/bin/rspec)
```

Comment:
Actual error is : 

```
/gems/packable-1.3.10/lib/packable/extensions/io.rb:86:in `pos': Illegal seek @ rb_io_tell - <STDOUT> (Errno::ESPIPE)

```

Please verify it.

Reply:
Yes, this was part of the error message too.

Reply:
I don't see those lines.Here is trace for me : 

```
 daru git:(master) bundle exec rspec
.................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................*.....................................................................................................................................*............................................................*.......................................................................................................................................................................................................................................................................................................................................................*.........*....*...........................................*.................../home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/nmatrix-0.2.4/lib/nmatrix/monkeys.rb:49: warning: constant ::Fixnum is deprecated
................................................................................................................................*.........................F......................................**.................................................................................Rserve: no process found

R version 3.4.4 (2018-03-15) -- "Someone to Lean On"
Copyright (C) 2018 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

Rserv started in daemon mode.
/home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/rserve-client-0.3.5/lib/rserve/rexp/wrapper.rb:26: warning: constant ::Fixnum is deprecated
/home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/rserve-client-0.3.5/lib/rserve/rexp/wrapper.rb:56: warning: constant ::Fixnum is deprecated
./home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/rserve-client-0.3.5/lib/rserve/rexp/wrapper.rb:26: warning: constant ::Fixnum is deprecated
/home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/rserve-client-0.3.5/lib/rserve/rexp/wrapper.rb:56: warning: constant ::Fixnum is deprecated
...NOTE: Daru::Vector#only_valid is deprecated; use reject_values instead. It will be removed on or after 2016-10-01.
Daru::Vector#only_valid called from /usr/lib/ruby/2.5.0/delegate.rb:83.
...............................................................................................................................................................................................................................................*.............................................................................................................................*.........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................*.........*....................................................................................................................................................................................................................................*.........*........................................................................................................................................................................................................................................*.........*.........................................................................*................................................................................................................................

Traceback (most recent call last):
  9: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/simplecov-0.16.1/lib/simplecov/defaults.rb:27:in `block in <top (required)>'
  8: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/simplecov-0.16.1/lib/simplecov.rb:200:in `run_exit_tasks!'
  7: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/simplecov-0.16.1/lib/simplecov/configuration.rb:182:in `block in at_exit'
  6: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/simplecov-0.16.1/lib/simplecov/result.rb:48:in `format!'
  5: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/simplecov-html-0.10.2/lib/simplecov-html.rb:25:in `format'
  4: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/simplecov-html-0.10.2/lib/simplecov-html.rb:25:in `puts'
  3: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/simplecov-html-0.10.2/lib/simplecov-html.rb:25:in `puts'
  2: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/packable-1.3.10/lib/packable/extensions/io.rb:62:in `write_with_packing'
  1: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/packable-1.3.10/lib/packable/extensions/io.rb:86:in `pack_and_write'
/home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/packable-1.3.10/lib/packable/extensions/io.rb:86:in `pos': Illegal seek @ rb_io_tell - <STDOUT> (Errno::ESPIPE)
Traceback (most recent call last):
  19: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/bin/rspec:23:in `<main>'
  18: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/bin/rspec:23:in `load'
  17: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/rspec-core-3.8.0/exe/rspec:4:in `<top (required)>'
  16: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/runner.rb:45:in `invoke'
  15: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/runner.rb:71:in `run'
  14: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/runner.rb:87:in `run'
  13: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/runner.rb:110:in `run_specs'
  12: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/reporter.rb:76:in `report'
  11: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/reporter.rb:166:in `finish'
  10: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/reporter.rb:186:in `close_after'
   9: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/reporter.rb:169:in `block in finish'
   8: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/reporter.rb:200:in `notify'
   7: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/reporter.rb:200:in `each'
   6: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/reporter.rb:201:in `block in notify'
   5: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/formatters/base_text_formatter.rb:49:in `dump_pending'
   4: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/output_wrapper.rb:24:in `block (2 levels) in <class:OutputWrapper>'
   3: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/rspec-core-3.8.0/lib/rspec/core/output_wrapper.rb:24:in `puts'
   2: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/packable-1.3.10/lib/packable/extensions/io.rb:62:in `write_with_packing'
   1: from /home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/packable-1.3.10/lib/packable/extensions/io.rb:86:in `pack_and_write'
/home/sciruby/daru/vendor/bundle/ruby/2.5.0/gems/packable-1.3.10/lib/packable/extensions/io.rb:86:in `pos': Illegal seek @ rb_io_tell - <STDOUT> (Errno::ESPIPE)
```

Reply:
Um, I got something slightly different:
```
......................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................./home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/nmatrix-0.2.4/lib/nmatrix/monkeys.rb:49: warning: constant ::Fixnum is deprecated
.........................................................................................................................................................F......................................**................................................................................./home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rserve-client-0.3.5/lib/rserve/rexp/wrapper.rb:26: warning: constant ::Fixnum is deprecated
/home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rserve-client-0.3.5/lib/rserve/rexp/wrapper.rb:56: warning: constant ::Fixnum is deprecated
./home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rserve-client-0.3.5/lib/rserve/rexp/wrapper.rb:26: warning: constant ::Fixnum is deprecated
/home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rserve-client-0.3.5/lib/rserve/rexp/wrapper.rb:56: warning: constant ::Fixnum is deprecated
...NOTE: Daru::Vector#only_valid is deprecated; use reject_values instead. It will be removed on or after 2016-10-01.
Daru::Vector#only_valid called from /home/sumera/.rbenv/versions/2.5.1/lib/ruby/2.5.0/delegate.rb:83.
.....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................*................................................................................................................................

bundler: failed to load command: rspec (/home/sumera/.rbenv/versions/2.5.1/bin/rspec)
...: command not found

Errno::ESPIPE: Illegal seek @ rb_io_tell - <STDERR>
  /home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/packable-1.3.10/lib/packable/extensions/io.rb:86:in `pos'
  /home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/packable-1.3.10/lib/packable/extensions/io.rb:86:in `pack_and_write'
  /home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/packable-1.3.10/lib/packable/extensions/io.rb:62:in `write_with_packing'
  /home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/cli/exec.rb:81:in `abort'
  /home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/cli/exec.rb:81:in `rescue in kernel_load'
  /home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/cli/exec.rb:65:in `kernel_load'
  /home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/cli/exec.rb:28:in `run'
  /home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/cli.rb:463:in `exec'
  /home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
  /home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
  /home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
  /home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/cli.rb:27:in `dispatch'
  /home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
  /home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/cli.rb:18:in `start'
  /home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/exe/bundle:30:in `block in <top (required)>'
  /home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/friendly_errors.rb:124:in `with_friendly_errors'
  /home/sumera/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/exe/bundle:22:in `<top (required)>'
  /home/sumera/.rbenv/versions/2.5.1/bin/bundle:23:in `load'
  /home/sumera/.rbenv/versions/2.5.1/bin/bundle:23:in `<main>'

```
But the `failed to load command: rspec` didn't show up when I changed the version so I thought it is probably Daru breaking. Will look further into it.

Reply:
Actually all bundler related lines look irrelevent. Only focus on `packable/extensions/io.rb:86:in `pos': Illegal seek @ rb_io_tell - <STDOUT> (Errno::ESPIPE)`

Reply:
Yes, I have made those changes in my last PR. Kindly take a look.